runelite-client/src/main/java/net/runelite/client/plugins/itemstats/StatBoost.java

```diff
@@ -57,7 +70,14 @@ public StatChange effect(Client client)
 		int calcedDelta = heals(client);
 		if (boost && calcedDelta > 0)
 		{
-			max += calcedDelta;
+			if (capCalculator != null)
```

Comment:
I'm not a fan of complicating this parent class due to the implementation of CappedStatBoost. Instead, I'd rather contain all of the cap calculation in the `heals()` implementation of the child class.

eg. Use this implementation for CappedStatBoost and revert all the changes to this file.
```java
@Override
public int heals(Client client)
{
	final int current = getStat().getValue(client);
	final int max = getStat().getMaximum(client);
	final int delta = deltaCalculator.calculateDelta(max);
	final int cap = capCalculator.calculateDelta(max);

	if (delta + current <= max + cap)
	{
		return delta;
	}

	return max + cap - current;
}```